### PR TITLE
Fixing view error and renaming ContentTypeDeploymentStep

### LINKED ISSUE #1803: ContentTypesDeployment Step fails due to error in view
See below image:

image

This happens when you add the Content Types deployment step.
I'll send a fix shortly, including a better name. Although I understand the reasoning for the name, "content type" is easily confused with "content definition", so I suggest to rename the step (and relevant files) to just "content deployment step". I would further propose to merge the "all content" step with the "content" step, from where the user can choose to export content of all types, or of a certain type, or maybe even by specifying a query.

### PR DESCRIPTION
**Pull request by sfmskywalker**
Also some minor code cleanup.

Fixes #1803

### CODE REVIEW

**Comment by sfmskywalker** [comment]
@sebastienros I don't understand why travis is failing - it works on my machine :trollface:

**Comment by jtkech** [comment]
I already have the same problem with the functional tests, maybe 15s is a little short.

    Timeout - Async callback was not invoked within the 15000ms timeout specified by jest.setTimeout.

Then when it checks the page content you are still in the create tenant page.

Maybe you just have to retry it by pushing a fake change ;)

**Comment by sebastienros** [comment]
The failed functional tests have nothing to do with timeouts, I see actual error messages from the app.

**Comment by jtkech** [comment]
Ah, okay.

File: src/OrchardCore.Modules/OrchardCore.Deployment/Views/Items/ContentDeploymentStep.Fields.Summary.cshtml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 @model dynamic
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Views/Items/ContentDeploymentStep.Fields.Summary.cshtml]: Remove dynamic, use the correct view model

File: src/OrchardCore.Modules/OrchardCore.Deployment/Views/Items/ContentDeploymentStep.Fields.Thumbnail.cshtml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+@model dynamic
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Views/Items/ContentDeploymentStep.Fields.Thumbnail.cshtml]: Remove dynamic, use the correct view model

**Comment by sebastienros** [comment]
Maybe one of the recipes is using this step and it shows an issue. Try all the recipes.

**Comment by jtkech** [comment]
I could repro the issue by doing a blog setup.

I could fix it by re-using the setters in `ContentDefinitionStep.cs` line 87 and 88.

**Comment by sebastienros** [comment]
👏 thank you functional tests

File: src/OrchardCore.Modules/OrchardCore.Deployment/Views/Items/ContentDeploymentStep.Fields.Summary.cshtml (hunk 1 above)
Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Deployment/Views/Items/ContentDeploymentStep.Fields.Summary.cshtml]: This should be `ShapeViewModel<ContentDeploymentStep>`, then you can remove the cast too.

**Comment by jtkech** [comment]
Yes you're right 👍.

Most of the time i redo a setup for nothing and i don't do it just when it should have been done :)

**Comment by sfmskywalker** [comment]
Shit, you're right. I can indeed reproduce the error with the Blog recipe 👍

**Comment by sfmskywalker** [comment]
@jtkech 

>I could fix it by re-using the setters in ContentDefinitionStep.cs line 87 and 88.

What do you mean with re-using setters? 🤔

**Comment by sfmskywalker** [comment]
Ow wait - I removed the setters there - you mean, if you add them back it works. I'll give it a try, but you;re probably right. Thank you SonarLint 😡

**Comment by sfmskywalker** [comment]
Thank you @jtkech and @sebastienros for fixing it - probably saved me a lot of time.

**Pull request merged**